subworkflows/local/lr_preprocessing.nf

```diff
@@ -0,0 +1,77 @@
+/*
+ * LR_PREPROCESSING: Preprocessing and QC for long reads
+ */
+
+include { NANOPLOT as NANOPLOT_RAW                              } from '../../modules/nf-core/nanoplot/main'
+include { NANOPLOT as NANOPLOT_FILTERED                         } from '../../modules/nf-core/nanoplot/main'
+include { NANOLYSE                                              } from '../../modules/nf-core/nanolyse/main'
+include { PORECHOP_ABI                                          } from '../../modules/nf-core/porechop/abi/main'
+include { FILTLONG                                              } from '../../modules/local/filtlong'
+
+
+workflow LR_PREPROCESSING {
+    take:
+    ch_raw_long_reads         // [ [meta] , fastq] (mandatory)
+    ch_short_reads            // [ [meta] , fastq1, fastq2] (mandatory)
+    ch_nanolyse_db            // [fasta]
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    NANOPLOT_RAW (
+        ch_raw_long_reads
+    )
+
+    ch_versions = ch_versions.mix(NANOPLOT_RAW.out.versions.first())
+
+    ch_long_reads = ch_raw_long_reads
+                    .map {
+                        meta, reads ->
+                            def meta_new = meta - meta.subMap('run')
+                        [ meta_new, reads ]
+                    }
+
+    if ( !params.assembly_input ) {
+        if (!params.skip_adapter_trimming) {
+            PORECHOP_ABI (
+                ch_raw_long_reads
+            )
```

Comment:
Here is where we want the if else, e.g.

```suggestion
        if (!params.skip_adapter_trimming) {
            if (params.longread_preprocessing_tool == 'porechop_abi') {
                PORECHOP_ABI (
                    ch_raw_long_reads
                )
             } else (params.longread_preprocessing_tool == 'porechop')
                <...>
```

Reply:
great suggestion. what if me make it a comma-separated string, where the user can specify all LR-preprocessing tools to use?